tests/e2e/standard/enketo/pregnancy.wdio-spec.js

```diff
@@ -0,0 +1,143 @@
+const contactPage = require('../../../page-objects/standard/contacts/contacts.wdio.page');
+const reportsPage = require('../../../page-objects/reports/reports.wdio.page');
+const analyticsPage = require('../../../page-objects/analytics/analytics.wdio.page');
+const genericForm = require('../../../page-objects/forms/generic-form.wdio.page');
+const pregnancyForm = require('../../../page-objects/standard/forms/pregnancy.wdio.page');
+
+describe('New pregnancy', () => {
+  const places = placeFactory.generateHierarchy();
+  const healthCenter = places.find(place => place.type === 'health_center');
+  const user = userFactory.build({ place: healthCenter._id, roles: ['district_admin'] });
+  const pregnantWoman1 = 'Woman1';
+  const pregnantWoman2 = 'Woman2';
+
+  before(async () => { 
+    await utils.saveDocs([...places]);
+    await utils.createUsers([user]);
+    await loginPage.cookieLogin();
+    await commonPage.goToPeople(healthCenter._id);
+    
+    // Create Woman1 - webapp
+    await contactPage.contactPageDefault.createNewPerson();
+    await newPersonPage.fillDefaultContactForm();
+    await genericForm.submitForm();
+
+    // Createn Woman2 - SMS N form
+    const messageNewWoman = `N ${pregnantWoman2}`;    
+    await gatewayApiUtils.api.postMessage({
+      id: 'N-id',
+      from: user.phone,
+      content: messageNewWoman
+    }); 
+  });
+
+  it('Submit new pregnancy - Woman1 - webapp', async () => {
+    const note = 'Test note - pregnant woman';
+    await contactPage.contactPageDefault.selectLHSRowByText(pregnantWoman1);
+    const medicIDW1 = await contactPage.contactPageDefault.getContactMedicID();
+    await contactPage.contactPageDefault.createNewAction('New Pregnancy');
+    
+    await pregnancyForm.selectKnowLMP();
+    await pregnancyForm.selectAproxLMP(pregnancyForm.APROX_LMP.b7To8Months);
+    
+    expect(await (await pregnancyForm.getEstDeliveryDate()).isDisplayed()).to.be.true;
+ 
+    await genericForm.nextPage();
+    const riskFactors = await pregnancyForm.selectAllRiskFactors();
+    await genericForm.nextPage();
+    const dangerSigns = await pregnancyForm.selectAllDangerSigns();
+    await genericForm.nextPage();
+    await pregnancyForm.setNote(note);
+    await genericForm.nextPage();
+
+    expect((await pregnancyForm.riskFactorsSummary()).length).to.equal(riskFactors.length);
+    expect((await pregnancyForm.dangerSignsSummary()).length).to.equal(dangerSigns.length);
+    const followUpSMS = await pregnancyForm.getFollowUpSMS();
+    expect(followUpSMS).to.include(pregnantWoman1);
+    expect(followUpSMS).to.include(medicIDW1);
+    expect(followUpSMS).to.include(note);
+
+    await genericForm.submitForm();
+    await commonPage.waitForPageLoaded();
+    
+    expect(await (await contactPage.pregnancyCard()).isDisplayed()).to.be.true;
+    expect(await contactPage.getPregnancyCardRisk()).to.equal('High risk');
+  });
+
+  it('Verify pregnancy report - Woman1 - webapp', async () => {
```

Comment:
This is not really a problem, but more of an observation.  Typically, with automated tests, it is recommended that each test be self-contained and atomic (and not rely on any other test to have run first).  In the case of this test, I think it needs the previous test to run first.

I don't think this is a huge problem if this is part of the general pattern we want to follow for these e2e tests.  However, if we are not already committed to this pattern, I would recommend we avoid dependencies between tests since it can make debugging issues a lot more tricky (and makes the idea of eventually running these test in parallel impossible...).

Reply:
That is a really good point, I haven't thought about it.

In the last individual test of this group I check the `targets` and those values include the pregnancy registered by `webapp` and the pregnancy registered by `SMS`, so I will not be able to separate anything and the five `it` will need to be all together in a single one.

Do you think that it will be a better idea to have it all in just one individual test?

I agree with you that no other test should depend on the result of another test, but I also like the idea to know what is being tested in every step of the process, and at the end every file has the `describe` that includes all the individual tests. 
What do you think?

Reply:
> Do you think that it will be a better idea to have it all in just one individual test?

I am interested to hear what @ngaruko thinks, but as nasty as it would look, I think it would be better to have one big test, then to have inter-dependencies between tests.

One interesting alternative would be to centralize the part that is needed for each test (e.g. submitting the pregnancy form) into a `before` method that runs once for the `describe`.  Then you can still break out the various assertions into separate `it` functions to give it some better structure. Then you could also use nested `describe` blocks to layer in some more complexity (e.g. submit the regular pregnancy form in the outer block and then have an inner `describe` with a `before` that submits the sms form.  Then `it` functions in the inner `describe` can include assertions that require both forms to have been submitted...)

Reply:
I think it makes sense @jkuester @tatilepizs to keep each test independent and use the `before hooks` for pre-requisites. If the pre-requisite is not part of the actual test, the most efficient/stable way I have seen is to maybe try and use a non-UI way to achieve that , when possible, same we do when we create users or seed other docs needed for a test.

Reply:
Thank you both for the feedback!
I moved to just one individual test (`it`) the creation of the report, the verification of the contact cards, and the verification of the reports, which actually makes more sense.